ipa-core/src/net/test.rs

```diff
@@ -694,48 +816,52 @@ a0778c3e9960576cbef4312a3b7ca34137880fd588c11047bd8b6a8b70b5a151
 mod tests {
     use super::{get_test_certificate_and_key, TestConfigBuilder};
     use crate::{
+        config::NetworkConfig,
         helpers::HelperIdentity,
-        net::test::{Ports, TEST_CERTS, TEST_KEYS},
+        net::{
+            test::{Ports, FOUR_SHARDS, TEST_CERTS, TEST_KEYS},
+            ConnectionFlavor,
+        },
         sharding::{ShardIndex, ShardedHelperIdentity},
     };
 
+    fn assert_eq_configs<F: ConnectionFlavor>(nc1: &NetworkConfig<F>, nc2: &NetworkConfig<F>) {
+        let urls1: Vec<_> = nc1.vec_peers().into_iter().map(|p| p.url).collect();
+        let urls2: Vec<_> = nc2.vec_peers().into_iter().map(|p| p.url).collect();
+        assert_eq!(urls1, urls2);
+    }
+
     /// This simple test makes sure that testing networks are created properly.
     /// The network itself won't be excersized as that's tested elsewhere.
     #[test]
     fn create_4_shard_http_network() {
-        let ports: Vec<Ports> = vec![
-            Ports {
-                ring: [10000, 10001, 10002],
-                shards: [10005, 10006, 10007],
-            },
-            Ports {
-                ring: [10010, 10011, 10012],
-                shards: [10015, 10016, 10017],
-            },
-            Ports {
-                ring: [10020, 10021, 10022],
-                shards: [10025, 10026, 10027],
-            },
-            Ports {
-                ring: [10030, 10031, 10032],
-                shards: [10035, 10036, 10037],
-            },
-        ];
         // Providing ports and no https certs to keep this test fast
         let conf = TestConfigBuilder::default()
             .with_disable_https_option(true)
-            .with_ports_by_ring(ports)
+            .four_shards()
             .build();
 
         assert!(conf.disable_https);
         assert_eq!(conf.rings.len(), 4);
         assert_eq!(conf.shards.len(), 3);
-        let shards_2 = conf.get_shards_for_helper(HelperIdentity::TWO);
-        assert_eq!(shards_2.get_first_shard().config.port, Some(10006));
-        let second_helper_third_shard_configs = &shards_2.servers[2];
-        assert_eq!(second_helper_third_shard_configs.config.port, Some(10026));
-        let leader_ring_configs = &conf.leaders_ring().servers;
-        assert_eq!(leader_ring_configs[2].config.port, Some(10002));
+
+        let apps = conf.into_apps();
+        for (i, ports) in FOUR_SHARDS.iter().enumerate() {
+            for (j, port) in ports.ring.into_iter().enumerate() {
+                assert_eq!(apps[i * 3 + j].mpc_server.config.port, Some(port));
+                assert_eq_configs(
+                    &apps[i * 3].mpc_network_config,
+                    &apps[i * 3 + j].mpc_network_config,
```

Comment:
I remember you added a function to compute that index, would it work here?

Reply:
Since I'm enumerating over the ports it won't be straight forward. I don't want to iterate over the computed results since they could have a problem (perhaps a missing ring), I rather iterate over the "known" result.